infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServerCache.java

```diff
@@ -144,4 +132,42 @@ public void invalidateAll() {
     public int getSize() {
         return dataCache.asMap().size();
     }
+
+    @Builder
+    public static class DataCacheConfigurator {
+        @Default
+        @NonNull
+        private final Weigher<Long, ILogData> weigher = (addr, logData) -> {
+            if (logData.getStreams().size() > MAX_STREAM_THRESHOLD) {
+                log.warn("Number of streams in this data is higher that threshold {}." +
+                        "This may impact the server performance", MAX_STREAM_THRESHOLD);
+            }
+
+            return Math.addExact(logData.getTotalSize(), KEY_SIZE);
+        };
+
+        @NonNull
+        private final LogUnitServerConfig config;
+
+        @Default
+        @NonNull
+        private final RemovalListener<Long, ILogData> evictionListener = (address, entry, cause) -> {
+            if (log.isTraceEnabled()) {
+                log.trace("handleEviction: Eviction[{}]: {}", address, cause);
+            }
+        };
+
+        @NonNull
+        private final CacheLoader<Long, ILogData> cacheLoader;
+
+        public LoadingCache<Long, ILogData> buildDataCache() {
```

Comment:
If the purpose of this class is to have different cache loaders (other than Caffine), do you perhaps want to name this builder to explicitly say `buildCaffineDataCache`?

Reply:
Right now, I just want to be able to build the cache and provide my own loader function to it in tests. When we will need to bring another cache I will refactor accordingly, right now no need for that

Reply:
And yes, I have just renamed the method just to `dataCache()`
If you have better names to offer, I'm happy to use it